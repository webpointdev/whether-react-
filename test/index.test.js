import {Whether, Match, Else} from '../es';
import {create} from 'react-test-renderer';

test('one branch matched', () => {
    const component = create(
        <Whether matches={true}>
            <div />
        </Whether>
    );

    expect(component.toJSON()).toMatchSnapshot();
});

test('one branch with multiple children', () => {
    const component = create(
        <Whether matches={true}>
            <ul />
            <ol />
        </Whether>
    );

    expect(component.toJSON()).toMatchSnapshot();
});

test('one branch not matched', () => {
    const component = create(
        <Whether matches={false}>
            <div />
        </Whether>
    );

    expect(component.toJSON()).toMatchSnapshot();
});

test('one branch matched with function children', () => {
    const component = create(
        <Whether matches={true}>
            {() => <div />}
        </Whether>
    );

    expect(component.toJSON()).toMatchSnapshot();
});

test('if and else when matched', () => {
    const component = create(
        <Whether matches={true}>
            <div />
            <Else>
                <span />
            </Else>
        </Whether>
    );

    expect(component.toJSON()).toMatchSnapshot();
});

test('if and else with multiple children matched', () => {
    const component = create(
        <Whether matches={true}>
            <ul />
            <ol />
            <Else>
                <span />
            </Else>
        </Whether>
    );

    expect(component.toJSON()).toMatchSnapshot();
});

test('if and else when not matched', () => {
    const component = create(
        <Whether matches={false}>
            <div />
            <Else>
                <span />
            </Else>
        </Whether>
    );

    expect(component.toJSON()).toMatchSnapshot();
});

test('else with multiple children', () => {
    const component = create(
        <Whether matches={true}>
            <div />
            <Else>
                <ul />
                <ol />
            </Else>
        </Whether>
    );

    expect(component.toJSON()).toMatchSnapshot();
});

test('if and else when not matched with function children', () => {
    const component = create(
        <Whether matches={false}>
            <div />
            <Else>
                {() => <span />}
            </Else>
        </Whether>
    );

    expect(component.toJSON()).toMatchSnapshot();
});

test('multiple match', () => {
    const component = create(
        <Whether context={3}>
            <Match selector={i => i < 1}>
                <span />
            </Match>
            <Match selector={i => i > 2}>
                <div />
            </Match>
        </Whether>
    );

    expect(component.toJSON()).toMatchSnapshot();
});

test('match with multiple children', () => {
    const component = create(
        <Whether context={3}>
            <Match selector={i => i < 1}>
                <span />
            </Match>
            <Match selector={i => i > 2}>
                <ul />
                <ol />
            </Match>
        </Whether>
    );

    expect(component.toJSON()).toMatchSnapshot();
});

test('multiple match when none matched', () => {
    const component = create(
        <Whether context={3}>
            <Match selector={i => i < 1}>
                <div />
            </Match>
            <Match selector={i => i < 2}>
                <span />
            </Match>
        </Whether>
    );

    expect(component.toJSON()).toMatchSnapshot();
});

test('multiple match with else matched', () => {
    const component = create(
        <Whether context={3}>
            <Match selector={i => i < 1}>
                <div />
            </Match>
            <Else>
                <span />
            </Else>
        </Whether>
    );

    expect(component.toJSON()).toMatchSnapshot();
});

test('match with function children', () => {
    const component = create(
        <Whether context={3}>
            <Match selector={i => i > 1}>
                {() => <div />}
            </Match>
            <Else>
                {() => <span />}
            </Else>
        </Whether>
    );

    expect(component.toJSON()).toMatchSnapshot();
});

test('else with function children', () => {
    const component = create(
        <Whether context={3}>
            <Match selector={i => i < 1}>
                {() => <div />}
            </Match>
            <Else>
                {() => <span />}
            </Else>
        </Whether>
    );

    expect(component.toJSON()).toMatchSnapshot();
});