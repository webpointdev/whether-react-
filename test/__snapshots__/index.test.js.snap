// Jest Snapshot v1, https://goo.gl/fbAQLP

exports[`else with function children 1`] = `<span />`;

exports[`else with multiple children 1`] = `<div />`;

exports[`if and else when matched 1`] = `<div />`;

exports[`if and else when not matched 1`] = `<span />`;

exports[`if and else when not matched with function children 1`] = `<span />`;

exports[`if and else with multiple children matched 1`] = `
Array [
  <ul />,
  <ol />,
]
`;

exports[`match with function children 1`] = `<div />`;

exports[`match with multiple children 1`] = `
Array [
  <ul />,
  <ol />,
]
`;

exports[`multiple match 1`] = `<div />`;

exports[`multiple match when none matched 1`] = `null`;

exports[`multiple match with else matched 1`] = `<span />`;

exports[`one branch matched 1`] = `<div />`;

exports[`one branch matched with function children 1`] = `<div />`;

exports[`one branch not matched 1`] = `null`;

exports[`one branch with multiple children 1`] = `
Array [
  <ul />,
  <ol />,
]
`;
